akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,471 @@
+
+.. figure:: ../images/persistence-drop-event-serializer.png
+   :align: center
+
+   The serializer is aware of the old event types that need to be skipped (**O**), and can skip deserializing them alltogether
+   by simply returning a "tombstone" (**T**), which the EventAdapter converts into an empty EventSeq.
+   Other events (**E**) can simply be passed through.
+
+The serializer detects that the string manifest points to a removed event type and skips attempting to deserialize it:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#string-serializer-skip-deleved-event-by-manifest
+
+The EventAdapter we implemented is aware of ``EventDeserializationSkipped`` events (our "Tombstones"),
+and emits and empty ``EventSeq`` whenever such object is encoutered:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#string-serializer-skip-deleved-event-by-manifest-adapter
+
+.. _detach-domain-from-data-model-scala:
+
+Detach domain model from data model
+-----------------------------------
+
+**Situation:**
+You want to separate the application model (often called the "*domain model*") completely from the models used to
+persist the corresponding events (the "*data model*"). For example because the data representation may change
+independently of the domain model.
+
+Another situation where this technique may be useful is when your serialization tool of choice requires generated
+classes to be used for serialization and deserialization of objects, like for example `Google Protocol Buffers`_ do,
+yet you do not want to leak this implementation detail into the domain model itself, which you'd like to model as
+plain Scala case classes.
+
+**Solution:**
+In order to detach the domain model, which is often represented using pure scala (case) classes, from the data model
+classes which very often may be less user-friendly yet highly optimised for throughput and schema evolution
+(like the classes generated by protobuf for example), it is possible to use a simple EventAdapter which maps between
+these types in a 1:1 style as illustrated below:
+
+.. figure:: ../images/persistence-detach-models.png
+   :align: center
+
+   Domain events (**A**) are adapted to the data model events (**D**) by the ``EventAdapter``.
+   The data model can be a format natively understood by the journal, such that it can store it more efficiently or
+   include additional data for the event (e.g. tags), for ease of later querying.
+
+We will use the following domain and data models to showcase how the separation can be implemented by the adapter:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#detach-models
+
+The :class:`EventAdapter` takes care of converting from one model to the other one (in both directions),
+alowing the models to be completely detached from each other, such that they can be optimised independently
+as long as the mapping logic is able to convert between them:
+
```

Comment:
Perhaps also mention that same thing can be done in a serializer instead of event adapter if the end result is bytes. E.g. protobuf, the serializer would map between domain model classes and bytes via generated protobuf builders.